File: Java/268. Missing Number.java
/**
 * https://leetcode.com/problems/missing-number/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    public int missingNumber(int[] nums) {
        int length = nums.length;
        int sum = ((length * (length + 1)) / 2);
        for (int num : nums)
            sum -= num;
        return sum;
    }
}

File: Kotlin/268. Missing Number.kt
/**
 * https://leetcode.com/problems/missing-number/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    fun missingNumber(nums: IntArray): Int {
        val len = nums.size
        var sum = (len * (len + 1)) / 2
        nums.forEach {
            sum -= it
        }
        return sum
    }
}